src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueStateMachineTests.cs

```diff
@@ -1099,715 +1094,3902 @@ .maxstack  3
+    IL_008c:  ldarg.0
+    IL_008d:  ldflda     ""System.Runtime.CompilerServices.AsyncTaskMethodBuilder C.<F>d__4.<>t__builder""
+    IL_0092:  ldloc.3
+    IL_0093:  call       ""void System.Runtime.CompilerServices.AsyncTaskMethodBuilder.SetException(System.Exception)""
+    IL_0098:  nop
+    IL_0099:  leave.s    IL_00af
+  }
+  IL_009b:  ldarg.0
+  IL_009c:  ldc.i4.s   -2
+  IL_009e:  stfld      ""int C.<F>d__4.<>1__state""
+  IL_00a3:  ldarg.0
+  IL_00a4:  ldflda     ""System.Runtime.CompilerServices.AsyncTaskMethodBuilder C.<F>d__4.<>t__builder""
+  IL_00a9:  call       ""void System.Runtime.CompilerServices.AsyncTaskMethodBuilder.SetResult()""
+  IL_00ae:  nop
+  IL_00af:  ret
+}
+");
+        }
+
+        [Fact]
+        public void UpdateAsync_Await_Remove_TryBlock()
+        {
+            var source0 = MarkedSource(@"
+using System.Threading.Tasks;
+
+class C
+{
+    static void Start() {}
+    static Task M1() => null;
+    static Task M2() => null;
+    static Task M3() => null;
+    static Task M4() => null;
+    static Task M5() => null;
+    static void End() {}
+
+    static async Task F() 
+    {
+        Start();
+        <N:0>await M1()</N:0>;
+        try
+        {
+            <N:1>await M2()</N:1>;
+            <N:2>await M3()</N:2>;
+        }
+        catch
+        {
+            <N:3>await M4()</N:3>;
+            <N:4>await M5()</N:4>;
+        }
+        End();
+    }
+}");
+            var source1 = MarkedSource(@"
+using System.Threading.Tasks;
+
+class C
+{
+    static void Start() {}
+    static Task M1() => null;
+    static Task M2() => null;
+    static Task M3() => null;
+    static Task M4() => null;
+    static Task M5() => null;
+    static void End() {}
+
+    static async Task F() 
+    {
+        Start();
+        try
+        {
+            <N:1>await M2()</N:1>;
+        }
+        catch
+        {
+            <N:2>await M4()</N:2>;
+        }
+        End();
+    }
+}");
+            var compilation0 = CreateCompilationWithMscorlib45(new[] { source0.Tree }, new[] { SystemCoreRef, CSharpRef }, options: ComSafeDebugDll);
+            var compilation1 = compilation0.WithSource(source1.Tree);
+
+            var v0 = CompileAndVerify(compilation0);
+            v0.VerifyDiagnostics();
+            var md0 = ModuleMetadata.CreateFromImage(v0.EmittedAssemblyData);
+
+            var f0 = compilation0.GetMember<MethodSymbol>("C.F");
+            var f1 = compilation1.GetMember<MethodSymbol>("C.F");
+
+            var generation0 = EmitBaseline.CreateInitialBaseline(md0, v0.CreateSymReader().GetEncMethodDebugInfo);
+
+            var diff1 = compilation1.EmitDifference(
+                 generation0,
+                 ImmutableArray.Create(
+                     SemanticEdit.Create(SemanticEditKind.Update, f0, f1, GetSyntaxMapFromMarkers(source0, source1), preserveLocalVariables: true)));
+
+            v0.VerifyIL("C.<F>d__7.System.Runtime.CompilerServices.IAsyncStateMachine.MoveNext", @"
```

Comment:
I was going to ask you to use a raw string literal here, so I could copy the IL into SharpLab and convert it to C# to understand it, but apparently that doesn't work :P